src/protocols/lido.ts

```diff
@@ -0,0 +1,152 @@
+    vault: IVaultInfo,
+    userKey: anchor.web3.PublicKey
+  ): Promise<{ txs: anchor.web3.Transaction[]; input: Buffer }> {
+    // Handle payload input here
+    const inputLayout = struct([nu64("amount")]);
+    let payload = Buffer.alloc(PAYLOAD_SIZE);
+    inputLayout.encode(
+      {
+        amount: new anchor.BN(params.depositAmount),
+      },
+      payload
+    );
+
+    // Handle transaction here
+
+    let preInstructions = [] as anchor.web3.TransactionInstruction[];
+
+    // Maybe better to do in navigator?
+    const bufferArray = [lido.LIDO_ADDRESS.toBuffer(), Buffer.from("reserve_account")];
+    const [reserveAccount] = anchor.web3.PublicKey.findProgramAddressSync(bufferArray, lido.LIDO_PROGRAM_ID);
+    const bufferArrayMint = [lido.LIDO_ADDRESS.toBuffer(), Buffer.from("mint_authority")];
+    const [mintAuthority] = anchor.web3.PublicKey.findProgramAddressSync(bufferArrayMint, lido.LIDO_PROGRAM_ID);
+
+    const recipientStSolAddress = await getAssociatedTokenAddress(vault.shareMint, userKey);
+
+    // TVerify if this requires a check here
+    preInstructions.push(await createATAWithoutCheckIx(userKey, vault.shareMint));
+
+    const remainingAccounts = [
+      { pubkey: lido.LIDO_ADDRESS, isSigner: false, isWritable: true }, // 0
+      { pubkey: userKey, isSigner: true, isWritable: true }, // 1 wallet.publicKey
+      { pubkey: recipientStSolAddress, isSigner: false, isWritable: true }, // 2
+      { pubkey: vault.shareMint, isSigner: false, isWritable: true }, // 3
+      { pubkey: reserveAccount, isSigner: false, isWritable: true }, // 4
+      { pubkey: mintAuthority, isSigner: false, isWritable: false }, // 5
+      { pubkey: TOKEN_PROGRAM_ID, isSigner: false, isWritable: false }, // 6
+      { pubkey: anchor.web3.SystemProgram.programId, isSigner: false, isWritable: false }, // 7
+    ];
+    const txDeposit = await this._gatewayProgram.methods
+      .deposit()
+      .accounts({
+        gatewayState: this._gatewayStateKey,
+        adapterProgramId: LIDO_ADAPTER_PROGRAM_ID,
+        baseProgramId: lido.LIDO_PROGRAM_ID,
+        activityIndex: await getActivityIndex(userKey),
+        gatewayAuthority: getGatewayAuthority(),
+      })
+      .preInstructions(preInstructions)
+      .remainingAccounts(remainingAccounts)
+      .transaction();
+
+    return { txs: [txDeposit], input: payload };
+  }
+
+  async withdraw(
+    params: WithdrawParams,
+    vault: IVaultInfo,
+    userKey: anchor.web3.PublicKey
+  ): Promise<{ txs: anchor.web3.Transaction[]; input: Buffer }> {
+    // Handle payload input here
+    const inputLayout = struct([nu64("amount"), u32("validatorIndex")]);
+    let payload = Buffer.alloc(PAYLOAD_SIZE);
+
+    const vaultInfo = vault as lido.VaultInfo;
+    const vaultInfoWrapper = new lido.VaultInfoWrapper(vaultInfo);
+
+    const heaviestValidator = vaultInfoWrapper.getHeaviestValidator();
+
+    const heaviestValidatorIndex = vaultInfo.validators.entries.findIndex((value) =>
```

Comment:
remove the heaviest validator logic over to `DappioWonderland/navigator`
in the Vault info wrapper itself